src/lava/proc/dense/models.py

```diff
@@ -16,17 +16,91 @@
 @implements(proc=Dense, protocol=LoihiProtocol)
 @requires(CPU)
 @tag('floating_pt')
-class PyDenseModel(PyLoihiProcessModel):
+class PyDenseModelFloat(PyLoihiProcessModel):
+    """Implementation of Conn Process with Dense synaptic connections in
+    floating point precision. This short and simple ProcessModel can be used
+    for quick algorithmic prototyping, without engaging with the nuances of a
+    fixed point implementation.
+    """
+    s_in: PyInPort = LavaPyType(PyInPort.VEC_DENSE, bool, precision=1)
+    a_out: PyOutPort = LavaPyType(PyOutPort.VEC_DENSE, float)
+    a_buff: np.ndarray = LavaPyType(np.ndarray, float)
+    weights: np.ndarray = LavaPyType(np.ndarray, float)
+    weight_exp: float = LavaPyType(float, float)
+    num_weight_bits: float = LavaPyType(float, float)
+    sign_mode: float = LavaPyType(float, float)
+
+    def run_spk(self):
+        # The a_out sent on a each timestep is a buffered value from dendritic
+        # accumulation at timestep t-1. This prevents deadlocking in
+        # networks with recurrent connectivity structures.
+        self.a_out.send(self.a_buff)
+        s_in = self.s_in.recv()
+        self.a_buff = self.weights[:, s_in].sum(axis=1)
+
+
+@implements(proc=Dense, protocol=LoihiProtocol)
+@requires(CPU)
+@tag('bit_accurate_loihi', 'fixed_pt')
+class PyDenseModelBitAcc(PyLoihiProcessModel):
+    """Implementation of Conn Process with Dense synaptic connections that is
+    bit-accurate with Loihi's hardware implementation of Dense, which means,
+    it mimics Loihi behaviour bit-by-bit.
+    """
+
     s_in: PyInPort = LavaPyType(PyInPort.VEC_DENSE, bool, precision=1)
     a_out: PyOutPort = LavaPyType(PyOutPort.VEC_DENSE, np.int32, precision=16)
-    # previously hidden var
+    a_buff: np.ndarray = LavaPyType(np.ndarray, np.int32, precision=16)
     weights: np.ndarray = LavaPyType(np.ndarray, np.int32, precision=8)
+    weight_exp: np.ndarray = LavaPyType(np.ndarray, np.int32, precision=4)
+    num_weight_bits: np.ndarray = LavaPyType(np.ndarray, np.int32, precision=3)
+    sign_mode: np.ndarray = LavaPyType(np.ndarray, np.int32, precision=2)
+
+    def __init__(self):
```

Comment:
This is missing type annotations (here only for the return type of `None`). But this is a more global issue, which should be addressed. Maybe not for the release... But we are piling up technical debt again.

Reply:
Yes, good idea -- we can come back and readdress this with all existing ProcModel code.